lib/Cake/Controller/Controller.php

```diff
@@ -785,6 +785,9 @@ public function redirect($url, $status = null, $exit = true) {
 			}
 		}
 
+		if ($status === null) {
+			$status = 302;
+		}
 		if ($status) {
```

Comment:
There was just one line that bothered me before:

    if (is_string($status)) {}

right before.
Which means that somewhere there could be a string, and it could be not resolved.